### PR TITLE
DataBase has been created

### PR DESCRIPTION
**Pull request by CrazyKamenwik**

### CODE REVIEW

File: BookOrderApi/BookOrderApi/AppDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        public AppDbContext()
+        {
+            Database.EnsureCreated();
+        }
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlServer(@"Server=localhost;Database=test;Trusted_Connection=True;");
+        }
+        //protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/AppDbContext.cs:24]: Get rid of comments

File: BookOrderApi/BookOrderApi/Controllers/BookController.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+using BookOrderApi.Entities;
+
+namespace BookOrderApi.Controllers
+{
+    [ApiController]
+    [Route("[Controller]")]
+    public class BookController : Controller
+    {
+        private readonly AppDbContext _dbcontext;
+        //private readonly HttpContextAccessor _httpcontext;
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/Controllers/BookController.cs:12]: Get rid of comments

File: BookOrderApi/BookOrderApi/Controllers/BookController.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+        public BookController(AppDbContext dbContext)
+        {
+            _dbcontext = dbContext;
+            //_httpcontext = httpContext;
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/Controllers/BookController.cs:17]: Get rid of comments

File: BookOrderApi/BookOrderApi/Entities/BookAuthor.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿//namespace BookOrderApi.Entities
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/Entities/BookAuthor.cs:1]: Get rid of comments

File: BookOrderApi/BookOrderApi/Entities/BookOrder.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿//namespace BookOrderApi.Entities
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/Entities/BookOrder.cs:1]: Get rid of comments

File: BookOrderApi/BookOrderApi/Entities/Order.cs [hunk 6]
```diff
@@ -3,7 +3,10 @@
     public class Order
     {
         public int Id { get; set; }
+        public int CustomerId { get; set; }
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/Entities/Order.cs:6]: Add navigation prop for Customer

File: BookOrderApi/BookOrderApi/desktop.ini [hunk 7]
```diff
@@ -0,0 +1,4 @@
+[ViewState]
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/desktop.ini:1]: Del this file

**Review by nantonov — CHANGES_REQUESTED** [review]

File: BookOrderApi/BookOrderApi/Entities/BookOrder.cs (hunk 5 above)
Comment (CrazyKamenwik) [reply BookOrderApi/BookOrderApi/Entities/BookOrder.cs:1]: > Get rid of comments

may be I should delete these files in case of their creation by ef core?

Comment (nantonov) [reply BookOrderApi/BookOrderApi/Entities/BookOrder.cs:1]: +

File: BookOrderApi/BookOrderApi/BookOrderApi.csproj [hunk 8]
```diff
@@ -6,10 +6,24 @@
     <ImplicitUsings>enable</ImplicitUsings>
   </PropertyGroup>
 
+  <ItemGroup>
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/BookOrderApi.csproj:9]: Delete this ItemGroup section

File: BookOrderApi/BookOrderApi/Controllers/BookController.cs [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        private readonly AppDbContext _dbcontext;
+
+        public BookController(AppDbContext dbContext)
+        {
+            _dbcontext = dbContext;
+        }
+
+        public IActionResult Index()
+        {
+            var data = _dbcontext.Books
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/Controllers/BookController.cs:21]: unnecessary

File: BookOrderApi/BookOrderApi/Controllers/BookController.cs [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+                .Include(x=>x.Authors)
+                .ToList();
+
+            var aut = _dbcontext.Authors
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/Controllers/BookController.cs:25]: Update naming

File: BookOrderApi/BookOrderApi/Controllers/BookController.cs [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    public class BookController : Controller
+    {
+        private readonly AppDbContext _dbcontext;
+
+        public BookController(AppDbContext dbContext)
+        {
+            _dbcontext = dbContext;
+        }
+
+        public IActionResult Index()
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/Controllers/BookController.cs:19]: Returns Authors, but controller named as Book

**Review by nantonov — CHANGES_REQUESTED** [review]

File: BookOrderApi/BookOrderApi/Controllers/BookController.cs [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            var books = _dbcontext.Books;
```
Comment (nantonov) [inline BookOrderApi/BookOrderApi/Controllers/BookController.cs:21]: ToList

**Review by nantonov — CHANGES_REQUESTED** [review]

**Review by nantonov — APPROVED** [review]

**Review by hromau — APPROVED** [review]

**Pull request merged**